overlord/snapstate/snapmgr.go

```diff
@@ -499,6 +511,59 @@ func (m *SnapManager) NextCatalogRefresh() time.Time {
 	return m.nextCatalogRefresh
 }
 
+func plugConnected(st *state.State, snapName, plugName string) bool {
+	conns, err := ifacerepo.Get(st).Connected(snapName, plugName)
+	return err == nil && len(conns) > 0
+}
+
+// refreshScheduleManaged returns true if the refresh schedule of the
+// device is managed by an external snap
+func (m *SnapManager) refreshScheduleManaged() bool {
+	var refreshScheduleStr string
+
+	tr := config.NewTransaction(m.state)
+	err := tr.Get("core", "refresh.schedule", &refreshScheduleStr)
+	if err != nil {
+		return false
+	}
+	if refreshScheduleStr != "managed" {
+		return false
+	}
+	snapStates, err := All(m.state)
```

Comment:
I think  everything from here and below belongs to an ifacestate helper that gets hooked into snapstate

Reply:
Thanks, I pushed a variant of your suggestion. It seems like a device property if the refreshes are managed or not so I moved it into devicestate and hook from there. But happy to go with your original suggestion if you feel strongly about it. I also moved to us snapdeclaration and started the store hints (in a separate branch in https://github.com/snapcore/snapd/compare/master...mvo5:refresh-candidates-managed?expand=1.

Reply:
no, that seem reasonable as well (devicestate)